### ISSUE TITLE
--strict does not convert WARNING during Lintables discovery into an ERROR

### ISSUE THREAD

**Issue by kpfleming** [ansible/ansible-lint#3062]
##### Summary

If the discovery of lintable files using Git fails (Git itself reports an error), ansible-lint produces a WARNING but `--strict` does not convert this into an ERROR. This can result in silent failures in CI jobs.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.13.1 using ansible 2.14.2
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

1. As user A, `git clone` a repository with content to be linted. Ensure that the resulting directory is readable by user B.
2. As user B, run `ansible-lint` in that directory.
3. `ansible-lint` will receive an error from Git while using it to discover the files in the directory; this will be reported as a WARNING, but the directory's contents will not be linted.
4. Add `--strict` to the `ansible-lint` invocation; there is no change in the result even though the help text for `--strict` says 'Return non-zero exit code on warnings as well as errors'.

##### Desired Behavior

This WARNING should be turned into an ERROR, as otherwise a CI job will report 'success' even though no files were linted.

##### Actual Behavior

```console (paste below)
lint-action: commands[4]> bash -c 'ansible-lint --strict --profile production -v workflow-support/*.yml src/roles/*'
INFO     Set ANSIBLE_LIBRARY=/github/home/.cache/ansible-compat/988886/modules:/github/home/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/github/home/.cache/ansible-compat/988886/collections:/tox/lint-action/tmp/collections:/tox/lint-action/tmp/collections
INFO     Set ANSIBLE_ROLES_PATH=/github/home/.cache/ansible-compat/988886/roles:/github/home/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
WARNING  Failed to discover lintable files using git: fatal: detected dubious ownership in repository at '/__w/ansible-systemd-networkd/ansible-systemd-networkd'
```

**Comment by ssbarnea** [comment]
I was about to close this as being outside the scope of warning mode, but after thinking a little bit more, I think what "why not". It is easy to implement it.

Can you check the pull request and confirm that it works well?

**Comment by kpfleming** [comment]
Indeed, looks very simple! I'll try to test this later today or tomorrow.
